### PR TITLE
[#1810] Fix filter glob user input

### LINKED ISSUE #1810: Filter by glob input does not allow user input
**Tell us about your environment**

* **RepoSense Version:** https://github.com/reposense/RepoSense/commit/11421c8394bdca3464957d550186715e9a94bf0a (bug seems to have appeared since a very long time ago)
* **OS and Version:** macOS 12.3.1
* **Web Browser and Version (if applicable):** Google Chrome 103.0.5060.53

**Please include the steps to reproduce the bug.**

1. Navigate to the RepoSense dashboard
2. Click on anyone's author contribution panel
3. Click on the "Filter by glob" input

**What was expected to happen?**
As a user, I should be able to provide user input to the "Filter by glob" input box.

**What actually happened? Please include a screenshot of the output.**
The input functions like a button, causing the report to refresh. It is not possible for the user to input anything into the "Filer by glob" input box.

**If possible, include the URL to your RepoSense report or log files (if any).**
https://dashboard-1780-pr-reposense-reposense.surge.sh/?search=&sort=groupTitle&sortWithin=title&since=2017-10-09&timeframe=commit&mergegroup=&groupSelect=groupByRepos&breakdown=false&tabOpen=true&tabType=authorship&tabAuthor=eugenepeh&tabRepo=reposense%2FRepoSense%5Bmaster%5D&authorshipIsMergeGroup=false&authorshipFileTypes=tests~docs~other~frontend~backend&authorshipIsBinaryFileTypeChecked=false&authorshipIsIgnoredFilesChecked=false

### PR DESCRIPTION
**Pull request by Punpun1643**
Fixes #1810

## Proposed commit message

```
The input glob previously does not allow user input. Once clicked, the
report will refresh and does not allow user to type in anything in the
input box.

Let's remove v-on:focus to stop listening for focus event and stop
triggering indicateSearchBar(), which causes an update to
selectedFileTypes when the user clicks on the input box, resulting in
the report refreshing.

Let's also update input#search to also trigger indicateSearchBar() on
change event. input#search should have the same behaviour when the user
presses the Enter key as when user clicks button#submit-button.
```

## Other information

N/A

### CODE REVIEW

File: frontend/src/views/v-authorship.vue [hunk 1]
```diff
@@ -159,13 +158,14 @@
           .ignored-segment
             .ignore-text File is ignored.
         pre.hljs.file-content(v-else-if="file.wasCodeLoaded", v-show="file.active")
-          v-segment-collection(v-bind:segments="file.segments", v-bind:path="file.path")
+          template(v-for="segment in file.segments")
```
Comment (gok99) [inline frontend/src/views/v-authorship.vue:161]: Looks like this PR is reverting some changes made in #1750. Was this unintentionally committed?

**Review by Zhou-Jiahao-1998 — APPROVED** [review]
LGTM. Glob filter works as expected.

**Comment by dcshzj** [comment]
Thanks for the fix @Punpun1643! It looks like it is working as intended. However, is it also possible to introduce a frontend test to test and ensure that this functionality works in the future as well?

**Comment by Punpun1643** [comment]
@dcshzj Alright sure, will work on it 👍

**Comment by damithc** [comment]
@Punpun1643 thanks for the PR on this. Let's try to merge this soon as a this a main feature in the UI that is currently not working at all.

**Comment by Zhou-Jiahao-1998** [comment]
@dcshzj Could you please help assist on this? Thank you!!

**Review by dcshzj — APPROVED** [review]

**Pull request merged**